scripts/OutputDifferences.Rmd

```diff
@@ -0,0 +1,129 @@
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+## Differences in Hector outputs
+
+Hello, this is `leeyabot`! `r emo::ji("robot")`
+
+The current pull request's outputs differ from `master` as follows:  
+
+
+``` {r differences, echo = FALSE, message = FALSE}
+# Load packages
+library(dplyr)
+library(ggplot2)
+library(hector)
+library(emo)
+library(here)
+
+# Set root directory
+BASE_DIR <- here::here()
+
+# Read in comp-data
+path <- file.path(BASE_DIR, "tests", "testthat", "compdata", "hector_comp.csv")
+comp_data <- read.csv(path)
+
+# Variables of interest
+vars <- c(ATMOSPHERIC_C(), GLOBAL_TEMP(), RF_TOTAL())
+
+# Organize data
+compdata <- comp_data %>% 
+    select(scenario, year, variable, value, units, version) %>%
+    filter(scenario == "hector_rcp45.ini") %>%
+    filter(variable %in% vars) %>%
+    mutate(scenario = "rcp45") %>%
+    arrange(variable)
+    
+# Read in files for just RCP 4.5 - v3
+rcp45 <- file.path(BASE_DIR, "inst", "input", "hector_rcp45.ini")
+
+# Run core and retrieve variable
+core <- newcore(rcp45)
+invisible(run(core))
+output <- fetchvars(core, core$strtdate:core$enddate, vars, "rcp45")
+output <- output %>% 
+    mutate(version = "latest") %>%
+    arrange(variable)
+
+# Find differences between versions
+differences <- compdata %>% mutate(diff = output$value - compdata$value)
+differences <- differences %>% select(scenario, year, variable, units, version, diff)
+```
+
+
+```{r summary-table, echo= FALSE}
+
+# Organize data
+output %>%
+    select(latest = value) -> new_values
+
+both_versions <- compdata %>%
+    select(year, variable, value) %>%
+    rename(v3.0.0 = value) %>%
+    cbind(new_values)
+
+# Define params
+years <- both_versions$year
+
+# Compute linear regression
+linear <- by(both_versions, both_versions$variable, function(x) lm (both_versions$latest~both_versions$v3.0.0))
+
+# Access r squared
+squares <- sapply(linear, summary)
+squares <- squares %>% 
+    as.data.frame() %>%
+    slice(8) %>%
+    t() -> squares
+
+colnames(squares) <- "R squared"
+
+# Calculate RMSE
+# Extract residuals
+atmos_res <- linear$atmos_c$residuals
+ftot_res <- linear$Ftot$residuals
+tgav_res <- linear$Tgav$residuals
+
+table <- cbind(atmos_res, ftot_res, tgav_res)
+
+# Define function
+RMSE <- function(res) {
+    r <- sqrt(mean(res^2))
+}
+
+# Find RMSE
+error <- as.data.frame(apply(table, MARGIN = 2, RMSE))
+rownames(error) <- c("atmos_c", "Ftot", "Tgav")
+
+# Find mean of data to normalize
+mean3 <- by(both_versions$v3.0.0, both_versions$variable, mean)
+mean3 <- as.data.frame(c(mean3[1], mean3[2], mean3[3]))
+
+# Divide RMSE by means
+NRMSE <- as.data.frame(error[,1] / mean3[,1])
+rownames(NRMSE) <- c("atmos_c", "Ftot", "Tgav")
```

Comment:
Instead of hard coding the strings "atmos_c", "Ftot", and "Tgav" consider using the R Hector functions `ATMOSPHERIC_C(), GLOBAL_TEMP(), RF_TOTAL()` that way if there are any changes in the strings on the cpp side of things this script won't fail.